test/modern_bpf/test_suites/syscall_enter_suite/mkdir_e.cpp

```diff
@@ -0,0 +1,40 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_mkdir
+TEST(SyscallEnter, mkdirE)
+{
+	auto evt_test = new event_test(PPME_SYSCALL_MKDIR_2_E);
+
+	evt_test->mark_single_64bit_syscall_as_interesting(__NR_mkdir);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	/* This is a mock mode just to check that the value is correctly caught. 
+	 * The syscall should fail!
+	 */
+	uint32_t mode = 7;
+	const char* path = NULL;
+	evt_test->assert_syscall_failure(syscall(__NR_mkdir, path, mode), "mkdir");
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* 1° Parameter: mode (type: PT_UINT32) */
+	evt_test->assert_u32_param(1, mode);
```

Comment:
The asserts too could be automatically fetched from the event table, right? I mean, we could parse the event table and know that we need to expect like N parameters of types X,Y,Z.

Hard part is knowing their values (eg: in this case `mode`).
WDYT?

Reply:
Yeah, this can be something that we do later if, when adding new fillers and new tests, we noticed that it is error-prone.
We can skip this now!